addon/components/search-box.js

```diff
@@ -1,57 +1,58 @@
-import Component from '@ember/component';
-import { isPresent } from '@ember/utils';
-import { task, timeout } from 'ember-concurrency';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { isNone } from '@ember/utils';
+import { timeout } from 'ember-concurrency';
+import {restartableTask} from "ember-concurrency-decorators";
 
 const DEBOUNCE_TIMEOUT = 250;
 
-export default Component.extend({
-  classNames: ['search-box'],
-  value: '',
-  liveSearch: true,
-  'data-test-search-box': true,
-  actions: {
-    update(value){
-      const liveSearch = this.get('liveSearch');
-      this.set('value', value);
-      if (liveSearch) {
-        this.get('searchTask').perform();
-      }
-    },
-    clear() {
-      this.set('value', '');
-      const clear = this.get('clear');
-      if (isPresent(clear)) {
-        clear();
-      }
-    },
-    focus() {
-      //place focus into the search box when search icon is clicked
-      this.element.querySelector('input[type="search"]').focus();
-    },
-  },
-  searchTask: task(function * () {
-    yield timeout(DEBOUNCE_TIMEOUT);
-    const value = this.get('value');
-    yield this.get('search')(value);
-  }).restartable(),
+export default class SearchBox extends Component {
+  @tracked value = '';
 
-  keyUp(event) {
-    const keyCode = event.keyCode;
-    const target = event.target;
+  get liveSearch() {
+    return isNone(this.args.liveSearch) ? true : this.args.liveSearch;
+  }
 
-    if ('search' !== target.type) {
-      return;
+  @action
+  update(value){
+    this.value = value;
+    if (this.liveSearch) {
+      this.searchTask.perform();
     }
+  }
 
-    if (13 === keyCode) {
-      this.get('searchTask').perform();
-      return;
+  @action
+  clear() {
+    this.value = '';
+    if (this.args.clear) {
+      this.args.clear();
     }
+  }
 
-    if(27 === keyCode) {
-      if (this.clear) {
-        this.clear();
+  @action
+  moveFocus(searchInput) {
+    // place focus into the search box when search icon is clicked
+    searchInput.focus();
+  }
+
+  @action
+  keyUp({ key }) {
+    switch (key) {
+    case 'Enter':
+      this.searchTask.perform();
+      break;
+    case 'Escape':
+      if (this.args.clear) {
```

Comment:
Feels like this should reset the input to empty even if no `clear` method is passed. Technically an enhancement I guess, so maybe skip it.

Reply:
agreed. the `clear()` action already does this. invoking that one instead...

Reply:
..done.